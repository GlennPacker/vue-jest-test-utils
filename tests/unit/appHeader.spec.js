import AppHeader from '@/components/AppHeader.vue';
import { mount } from '@vue/test-utils';

describe('AppHeader', () => {
    let wrapper;

    beforeEach(() => {
        wrapper = mount(AppHeader);
    });

    test('basic render', () => {
        expect(wrapper.html()).toMatchSnapshot();
    });

    test('logout button should show by default', () => {
        expect(wrapper.find('button').isVisible()).toBe(false);
    });

    test('logout button should show when loggedIn is true', async () => {
        wrapper.setData({ loggedIn: true });

        await wrapper.vm.$nextTick();
        expect(wrapper.find('button').isVisible()).toBe(true);
    });
});
