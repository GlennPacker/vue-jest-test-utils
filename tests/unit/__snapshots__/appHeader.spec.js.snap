// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppHeader basic render 1`] = `<div><button class="btn" style="display: none;">Logout</button></div>`;
